packages/taiko-client/driver/preconf_server/api.go

```diff
@@ -0,0 +1,95 @@
+package preconf_server
+
+import (
+	"net/http"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/labstack/echo/v4"
+)
+
+// PreconfBlockGroupStatus represents the status of a preconfirmation block group.
+type PreconfBlockGroupStatus string
+
+// PreconfBlockGroupStatus values.
+const (
+	StatusFinalBlockGroup   PreconfBlockGroupStatus = "finalBlockGroup"
+	StatusFinalPreconfGroup PreconfBlockGroupStatus = "finalPreconfGroup"
+)
+
+// PreconfTransactionsGroup represents a preconfirmation block group.
+type PreconfTransactionsGroup struct {
+	BlockID          uint64                  `json:"blockId"`
+	ID               uint64                  `json:"groupId"`
+	TransactionsList types.Transactions      `json:"transactions"`
+	GroupStatus      PreconfBlockGroupStatus `json:"groupStatus"`
```

Comment:
`PreconfBlockGroupStatus` to `PreconfTxGroupStatus`

Reply:
or name 'Tx Group' to 'batch'?

Reply:
> `PreconfBlockGroupStatus` to `PreconfTxGroupStatus`

changed